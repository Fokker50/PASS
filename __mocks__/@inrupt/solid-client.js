import { vi } from 'vitest';
import { mockSolidDatasetFrom, addMockResourceAclTo } from '@inrupt/solid-client';

export * from '@inrupt/solid-client';

const exampleUrl = 'https://example.com';

export const getPodUrlAll = vi.fn(() => Promise.resolve([]));
export const saveAclFor = vi.fn((url) =>
  Promise.resolve({
    ...addMockResourceAclTo(mockSolidDatasetFrom(url)),
    internal_accessTo: exampleUrl
  })
);
export const saveSolidDatasetAt = vi.fn((url) =>
  Promise.resolve(addMockResourceAclTo(mockSolidDatasetFrom(url)))
);
export const getSolidDataset = vi.fn((url) =>
  Promise.resolve(addMockResourceAclTo(mockSolidDatasetFrom(url)))
);
export const deleteFile = vi.fn(() => Promise.resolve());
