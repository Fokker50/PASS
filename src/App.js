import { useSession } from "./hooks";
import { createContext } from "react";
import Login from "./components/Login/Login";
import {
  UploadDocumentForm,
  FetchDocumentForm,
  DeleteDocumentForm,
} from "./components/Form";
import CrossPodQuery from "./components/Form/CrossPodQuery";
import CrossPodWrite from "./components/Form/CrossPodWrite";

const AppHeader = () => {
  return (
    <header>
      <h2>Getting Started with PASS</h2>
    </header>
  );
};

export const SessionContext = createContext(null);

const App = () => {
  const { session, handleLogin } = useSession();

  return (
    <>
      <AppHeader />
      <SessionContext.Provider value={{ session, handleLogin }}>
        <Login />
        <UploadDocumentForm />
        <FetchDocumentForm />
        <DeleteDocumentForm />
        <CrossPodQuery />
        <CrossPodWrite />
      </SessionContext.Provider>
    </>
  );
};

export default App;
