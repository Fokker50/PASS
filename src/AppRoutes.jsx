// React Imports
import React from 'react';
import { Routes, Route, Navigate } from 'react-router-dom';
// Page Imports
import UserSection from './routes/UserSection';
import Inbox from './routes/Inbox';
import Forms from './routes/Forms';

const AppRoutes = ({ loadingActive, loadMessages }) => {
  const restorePath = localStorage.getItem('restorePath');
  const path = restorePath ?? '/PASS/home';

  return (
    <Routes>
      <Route exact path="/PASS/" element={<Navigate to={path} replace />} />
      <Route path="/PASS/home" element={<UserSection loadingActive={loadingActive} />} />
      <Route path="/PASS/forms" element={<Forms />} />
      <Route path="/PASS/inbox" element={<Inbox loadMessages={loadMessages} />} />
      <Route path="*" element={<Navigate to={restorePath} replace />} />
    </Routes>
  );
};

export default AppRoutes;
