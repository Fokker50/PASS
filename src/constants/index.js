import RDF_PREDICATES from './rdf_predicates';
import INTERACTION_TYPES from './interaction_types';
import { SOLID_IDENTITY_PROVIDER, currentOidcIssuer } from './solid_provider';
import ENVIRONMENT from './environment';

export {
  RDF_PREDICATES,
  INTERACTION_TYPES,
  SOLID_IDENTITY_PROVIDER,
  currentOidcIssuer,
  ENVIRONMENT
};
