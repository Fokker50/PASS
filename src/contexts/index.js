import SelectUserContext from './selectUserContext';
import UserListContext from './userListContext';
import InboxMessageContext from './inboxMessageContext';

/**
 * The contexts moedule contains React Contexts that helps with user management
 * and selection
 *
 * @namespace contexts
 */

export { SelectUserContext, UserListContext, InboxMessageContext };
