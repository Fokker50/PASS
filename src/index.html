<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <title>Getting Started: Inrupt JavaScript Client Libraries</title>
    <link rel="stylesheet" href="./style.css" />
  </head>

  <body>
    <div id="root"></div>
    <script type="module" defer src="./index.js"></script>
  </body>
</html>
