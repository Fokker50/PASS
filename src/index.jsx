import React from 'react';
import ReactDOM from 'react-dom/client';
import { HashRouter as Router } from 'react-router-dom';
import { SessionProvider } from '@inrupt/solid-ui-react';
import { ThemeProvider } from '@mui/material/styles';
import CssBaseline from '@mui/material/CssBaseline';
import App from './App';
// import Layout from './layouts/Layouts';
import './style.css';
import theme from './theme';

const root = ReactDOM.createRoot(document.getElementById('root'));
root.render(
  <React.StrictMode>
    <Router>
      <SessionProvider>
        <CssBaseline />
        <ThemeProvider theme={theme}>
          <App />
        </ThemeProvider>
      </SessionProvider>
    </Router>
  </React.StrictMode>
);
