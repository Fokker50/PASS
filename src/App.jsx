import React, { useEffect, useMemo, useState } from 'react';
import { Routes, Route, Navigate } from 'react-router-dom';
import { useSession } from '@inrupt/solid-ui-react';
import { Login } from './components/Login';
import AppHeader from './components/AppHeader';
import Forms from './components/Forms';
import { UserSection } from './components/Users';
import { SelectUserContext, UserListContext } from './contexts';
import {
  getUsersFromPod,
  generateActivityTTL,
  generateUsersList,
  updateUserActivity
} from './utils';

/**
 * @typedef {import("./typedefs").userListObject} userListObject
 */

const App = () => {
  const { session } = useSession();
  const [selectedUser, setSelectedUser] = useState('');
  /** @type {[userListObject[], React.Dispatch<React.SetStateAction<userListObject[]>>]} */
  const [userList, setUserList] = useState([]);

  const selectedUserObject = useMemo(() => ({ selectedUser, setSelectedUser }), [selectedUser]);
  const userListObject = useMemo(() => ({ userList, setUserList }), [userList]);

  useEffect(() => {
    /**
     * A function that generates a Users container if logging in for the first
     * time and initalizes the list of users from Solid
     *
     * @function fetchData
     */
    async function fetchData() {
      await generateUsersList(session);
      await generateActivityTTL(session);
      await updateUserActivity(session);
      try {
        const listUsers = await getUsersFromPod(session);
        setUserList(listUsers);
      } catch {
        setUserList([]);
      }
    }

    if (session.info.isLoggedIn) {
      fetchData();
    }
  }, [session.info.isLoggedIn]);

  return (
    <SelectUserContext.Provider value={selectedUserObject}>
      <UserListContext.Provider value={userListObject}>
        <Routes>
          <Route
            exact
            path="/PASS/"
            element={
              session.info.isLoggedIn ? (
                <Navigate to="/PASS/home" />
              ) : (
                <>
                  <AppHeader isLoggedIn={session.info.isLoggedIn} />
                  <Login />
                </>
              )
            }
          />
          <Route
            path="/PASS/home/"
            element={
              session.info.isLoggedIn ? (
                <>
                  <AppHeader isLoggedIn={session.info.isLoggedIn} />
                  <UserSection />
                </>
              ) : (
                <Navigate to="/PASS/" />
              )
            }
          />
          <Route
            path="/PASS/forms/"
            element={
              session.info.isLoggedIn ? (
                <>
                  <AppHeader isLoggedIn={session.info.isLoggedIn} />
                  <Forms />
                </>
              ) : (
                <Navigate to="/PASS/" />
              )
            }
          />
          <Route path="*" element={<Navigate to="/PASS/" />} />
        </Routes>
      </UserListContext.Provider>
    </SelectUserContext.Provider>
  );
};

export default App;
