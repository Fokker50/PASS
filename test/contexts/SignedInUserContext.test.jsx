import { render, cleanup } from '@testing-library/react';
import React, { useContext } from 'react';
import { getPodUrlAll } from '@inrupt/solid-client';
import { expect, it, afterEach, describe, vi } from 'vitest';
import { SignedInUserContext, SignedInUserContextProvider } from '../../src/contexts';
import flushPromises from '../testHelpers';

const TestConsumer = () => {
  const { podUrl } = useContext(SignedInUserContext);

  return <div>{podUrl}</div>;
};

vi.mock('@inrupt/solid-client');

describe('SignedInUserContext', () => {
  afterEach(() => {
    cleanup();
    vi.clearAllMocks();
  });

  it('fetches the user properties and passes them to the consumer', async () => {
    getPodUrlAll.mockResolvedValue(['https://example.com/pod/']);
    const { container } = render(
      <SignedInUserContextProvider>
        <TestConsumer />
      </SignedInUserContextProvider>
    );
    await flushPromises();
    expect(container).toMatchSnapshot();
  });
});
