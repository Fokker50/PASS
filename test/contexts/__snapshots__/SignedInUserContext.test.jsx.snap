// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SignedInUserContext > fetches the user properties and passes them to the consumer 1`] = `
<div>
  <div>
    https://example.com/pod/
  </div>
</div>
`;
